File: server/iih_pub/iih.ci_pub/src/main/java/iih/ci/mrfp/other2mrfp/d/Other2mrfpAggDO.java
package iih.ci.mrfp.other2mrfp.d;

import xap.sys.appfw.orm.model.agg.BaseAggDO;
import xap.mw.core.data.BaseDO;
import xap.mw.coreitf.i.IAggDesc;
import xap.mw.coreitf.i.IBaseDO;
import iih.ci.mrfp.other2mrfp.d.desc.Other2mrfpAggDODesc;



/**
 * 病案首页其他信息
 */
public class Other2mrfpAggDO extends BaseAggDO {
	private static final long serialVersionUID = 1L;

	@Override
	public CiMrFpOtherDO getParentDO() {
		return ((CiMrFpOtherDO) super.getParentDO());
	}

	public void setParentDO(CiMrFpOtherDO headDO) {
		setParent(headDO);
	}

	public CiMrfpIntenCareDO[] getCiMrfpIntenCareDO() {
		IBaseDO[] dos = getChildren(CiMrfpIntenCareDO.class);
		if(dos==null) return null;
		CiMrfpIntenCareDO[] result=new CiMrfpIntenCareDO[dos.length];
		for(int i=0;i<dos.length;i++){
			result[i]=(CiMrfpIntenCareDO)dos[i];
		}
		return result;
	}
	
	public void setCiMrfpIntenCareDO(CiMrfpIntenCareDO[] dos) {
		setChildren(CiMrfpIntenCareDO.class, dos);
	}

	@Override
	public IAggDesc getAggDesc() {
		Other2mrfpAggDODesc desc = new Other2mrfpAggDODesc();

		return desc;
	}
	
	 public  BaseDO createParentDO() {
         return new CiMrFpOtherDO();
     }

     public  BaseDO createChildDO(String clzName) {
	     if (clzName.equals("iih.ci.mrfp.other2mrfp.d.CiMrfpIntenCareDO")) {
             return new CiMrfpIntenCareDO();
         }
         return null;
     }
}

File: client/iih_stub/iih.ci_stub/iih.ci.mrfp/other2mrfp/d/Other2mrfpAggDO.cs
using System;
using xap.mw.core.data;
using xap.mw.core.meta;

namespace iih.ci.mrfp.other2mrfp.d
{
	/// <summary>
	/// 病案首页其他信息DO
	/// </summary>
	public class Other2mrfpAggDO : BaseAggDO
	{	
	    private static string CIMRFPOTHERDO_CIMRFPINTENCAREDO="iih.ci.mrfp.other2mrfp.d.CiMrfpIntenCareDO";

        public Other2mrfpAggDO() {
            this.setParentDO(new CiMrFpOtherDO());
        }

  	    public CiMrFpOtherDO getParentDO() {
		    return ((CiMrFpOtherDO) base.ParentDO);
	    }      
	    
	    public void setParentDO(CiMrFpOtherDO headDO) {
		    base.ParentDO=headDO;
	    }
	    
	    public CiMrfpIntenCareDO[] getCiMrfpIntenCareDO() {		    
		    
		    BaseDO[] dos=getChildren(Type.GetType(CIMRFPOTHERDO_CIMRFPINTENCAREDO));
		    if (dos == null || dos.Length==0){return null;}
            CiMrfpIntenCareDO[] r = new CiMrfpIntenCareDO[dos.Length];
            for (int i = 0; i < dos.Length;i++ ){
                r[i] = dos[i] as CiMrfpIntenCareDO;
            }
	        return r;
		    
	    }
	    
	    public void setCiMrfpIntenCareDO(CiMrfpIntenCareDO[] dos) {
            if (dos == null || dos.Length == 0) return;
            setChildren(dos[0].GetType(), dos);
	    }


	    public override BaseDO createParentDO()
	    {
	        return new CiMrFpOtherDO();
	    }

        public override BaseDO createChildDO(string clzName)
	    {
            if (CIMRFPOTHERDO_CIMRFPINTENCAREDO == clzName)
	        {
                return new CiMrfpIntenCareDO();
	        }
	        return null;
	    }
	    
	    /// <summary>
        /// 获取子类全路径名数组
        /// </summary>
        /// <returns></returns>
        public override string[] getChildClassNames(){
            return  new string[]{"iih.ci.mrfp.other2mrfp.d.CiMrfpIntenCareDO"};
        }
	    public new static AggMeta GetMeta()
	    {
	        AggMeta meta = new AggMeta();
	        meta.OneToOneDOs = new string[]{};
	        return meta;
	    }
	    
	}
}
